disco/preprocess/select_timepoints2.py

```diff
@@ -166,166 +322,241 @@ def get_profile_data():
+                ag_series[bus]["condition"].append(CriticalCondition.MAX_NET_GENERATION)
+
+    total_gen = sum([dic["generation"] for bus, dic in ag_series.items() if "generation" in dic])
+    total_dem = sum([dic["demand"] for bus, dic in ag_series.items() if "demand" in dic])
     net_total_gen = total_gen - total_dem
-    if 'max_demand' in  critical_conditions: 
+    if CriticalCondition.MAX_DEMAND in critical_conditions:
         max_demand_idx = np.where(total_dem == np.amax(total_dem))[0].tolist()[0]
         head_critical_time_indices.append(max_demand_idx)
-    if 'min_demand' in  critical_conditions:    
+    if CriticalCondition.MIN_DEMAND in critical_conditions:
         min_demand_idx = np.where(total_dem == np.amin(total_dem))[0].tolist()[0]
         head_critical_time_indices.append(min_demand_idx)
-    if 'max_generation' in  critical_conditions:
+    if CriticalCondition.MAX_GENERATION in critical_conditions:
         max_gen_idx = np.where(total_gen == np.amax(total_gen))[0].tolist()[0]
         head_critical_time_indices.append(max_gen_idx)
-    if 'max_net_generation' in  critical_conditions:
+    if CriticalCondition.MAX_NET_GENERATION in critical_conditions:
         max_netgen_idx = np.where(net_total_gen == np.amax(net_total_gen))[0].tolist()[0]
         head_critical_time_indices.append(max_netgen_idx)
-    
-    critical_time_indices = [t  for bus, dic in ag_series.items() 
-                             for t in dic['critical_time_idx'] 
-                             if 'critical_time_idx' in dic.keys()]
+
+    critical_time_indices = [
+        t
+        for bus, dic in ag_series.items()
+        for t in dic["critical_time_idx"]
+        if "critical_time_idx" in dic
+    ]
     critical_time_indices += head_critical_time_indices
-    critical_time_indices = list(set(critical_time_indices))
-    critical_time_indices.sort()
-    destination_model_dir = os.path.join(destination_dir,'new_model')
-    os.makedirs(destination_model_dir, exist_ok=True)
+    critical_time_indices = sorted(set(critical_time_indices))
+    compression_rate = 0
     if recreate_profiles:
-        destination_profile_dir = os.path.join(destination_dir,'new_profiles')
-        os.makedirs(destination_profile_dir, exist_ok=True)
-        
-    for profile, val in profile_data.items():
-        if profile in used_profiles:
-            base_len = len(val['time_series'])
-            compression_rate = len(critical_time_indices)/base_len
-            if recreate_profiles:
-                data = val['time_series'][critical_time_indices]
-                new_profile_path = os.path.join(destination_profile_dir, f'{profile}.csv')
-                pd.DataFrame(data).to_csv(new_profile_path, index=False, header=False)
-            
-    reset_profile_data(used_profiles, critical_time_indices)
-    save_circuit(destination_model_dir)
-            
+        destination_profile_dir = destination_dir / "new_profiles"
```

Comment:
Does this feature need to be exposed to the user in the CLI? I'm not sure we even need it. The LoadShape.dss in the saved folder has all of the data, although it is in a less human-readable form.

Reply:
Maybe not! I added this with the assumption that we redirect to these files. But with the "Save" and "Action=..." feature, we no longer need it.